llpc/test/shaderdb/OOB_Check_Load_Array_lit.frag

```diff
@@ -0,0 +1,36 @@
+// This test checks if load access to a 2D (multidimensional) array using a runtime index
+// will be transformed to a out of bounds checks against all bounds of the array, moving the load into a conditionally executed BB.
+// The OOB check will possibly zero out the load results.
+
+// BEGIN_SHADERTEST
+/*
+; RUN: amdllpc -spvgen-dir=%spvgendir% -v %gfxip %s -enable-scratch-bounds-checks | FileCheck -check-prefix=SHADERTEST %s
+; SHADERTEST-LABEL: {{^// LLPC}} SPIRV-to-LLVM translation results
+; SHADERTEST: alloca [2048 x [2 x <4 x float>]], align 16, addrspace(5)
+; SHADERTEST: load i32, i32 addrspace(7)* getelementptr inbounds (<{ i32 }>, <{ i32 }> addrspace(7)* @0, i32 0, i32 0), align 4
+; SHADERTEST: load i32, i32 addrspace(7)* getelementptr inbounds (<{ i32 }>, <{ i32 }> addrspace(7)* @0, i32 0, i32 0), align 4
+; SHADERTEST: load <4 x float>, <4 x float> addrspace(64)* @1, align 16
+; SHADERTEST: getelementptr [2048 x [2 x <4 x float>]], [2048 x [2 x <4 x float>]] addrspace(5)* %{{.*}}, i32 0, i32 %{{.*}}, i32 %{{.*}}
+; SHADERTEST: %[[B:[0-9]+]] = icmp ult i32 %[[A:[0-9]+]], 2048
+; SHADERTEST: %[[D:[0-9]+]] = icmp ult i32 %[[C:[0-9]+]], 2
+; SHADERTEST: %[[E:[0-9]+]] = and i1 %[[B]], %[[D]]
+; SHADERTEST: br i1 %[[E]], label %{{.*}}, label %{{.*}}
```

Comment:
Can you save the BB names and match against them? This will make it easier to tell which instructions are in which BBs. The test description mentions different BBs, so the test should cover that.